tests/foreman/api/test_environment.py

```diff
@@ -33,38 +25,46 @@ def test_positive_create_1(self, name):
         @Feature: Environment
 
         """
-        # Create an environment and validate the returned data.
-        env = entities.Environment(name=name).create()
-        self.assertEqual(env.name, name)
-
-        # Let's double-check by reading the environment.
-        env = env.read()
-        self.assertEqual(env.name, name)
-
-    @data(
-        gen_string('alpha', 256),
-        gen_string('numeric', 256),
-        gen_string('alphanumeric', 256),
-        gen_string('cjk', random.randint(1, 255)),
-        gen_string('latin1', random.randint(1, 255)),
-        gen_string('utf8', random.randint(1, 255)),
-    )
-    def test_negative_create_1(self, name):
+        for name in (
+                gen_string(str_type)
+                for str_type in ('alpha', 'numeric', 'alphanumeric')):
+            with self.subTest(name):
+                env = entities.Environment(name=name).create()
+                self.assertEqual(env.name, name)
+                env = env.read()  # check sat isn't blindly handing back data
+                self.assertEqual(env.name, name)
+
+    def test_negative_create_1(self):
         """@Test: Create an environment and provide an invalid name.
 
         @Assert: The server returns an error.
 
         @Feature: Environment
 
-        In the context of this test, an "invalid name" is one that it either
-        too long or contains illegal characters. The API returns this error
-        message when an illegal character is submitted:
+        In this test, an "invalid name" is one that is too long.
+
+        """
+        for name in invalid_names_list():
+            with self.subTest(name):
+                with self.assertRaises(HTTPError):
+                    entities.Environment(name=name).create()
+
+    def test_negative_create_2(self):
+        """@Test: Create an environment and provide an invalid name.
```

Comment:
s/invalid/illegal. Eventually we may import test cases to a test case management tool and it will help if test case descriptions are different.